File: prototypeLeger/app/src/main/java/com/example/scrabbleprototype/objects/Themes.kt
package com.example.scrabbleprototype.objects

import com.example.scrabbleprototype.R
import com.example.scrabbleprototype.model.Item

object Themes {
    var appThemes = arrayListOf<String>()
    val boardItems = arrayListOf<Item>()
    val chatItems = arrayListOf<Item>()

    val defaultItem = Item("Par défaut", 0, R.color.light_green, "Composé de vert clair et de turquoise")

    init {
        appThemes.add("Default")
        appThemes.add("Dark")
        appThemes.add("Blue")
        appThemes.add("Orange")
        boardItems.add(Item("Gradients animés", 300, R.drawable.gradient_animation))
        boardItems.add(Item("Tartan", 100, R.drawable.tartan))
        boardItems.add(Item("Galaxie", 500, R.drawable.galaxy))

        chatItems.add(Item("Gradients animés", 100, R.drawable.gradient_animation))
        chatItems.add(Item("Galaxie", 100, R.drawable.galaxy))
    }

    fun getBoard(name: String): Item? {
        return boardItems.find { it.name == name }
    }
    fun getChat(name: String): Item? {
        return chatItems.find { it.name == name }
    }
}

File: client/src/app/classes/themes.ts
import { Item, Language, UserPreferences } from '@common/user-preferences';
import { galaxy, gradient, tartan } from '@common/themePics';
export class Themes {
    appThemes: string[];
    defaultItem: Item;
    boardItems: Item[];
    chatItems: Item[];

    constructor() {
        this.appThemes = ['Default', 'Dark', 'Blue', 'Orange'];
        this.defaultItem = new Item('Par défaut', 0, '', 'Composé de vert clair et de turquoise');
        this.boardItems = [
            {
                name: 'Gradients animés',
                price: 75,
                theme: gradient,
                description: '',
            },
            {
                name: 'Tartan',
                price: 50,
                theme: tartan,
                description: '',
            },
            {
                name: 'Galaxie',
                price: 50,
                theme: galaxy,
                description: '',
            },
        ];

        this.chatItems = [
            {
                name: 'Gradients animés',
                price: 75,
                theme: gradient,
                description: '',
            },
            {
                name: 'Galaxie',
                price: 50,
                theme: galaxy,
                description: '',
            },
        ];
    }

    getBoard(name: string): Item {
        return this.boardItems.find((curTheme) => curTheme.name === name) as Item;
    }

    getChat(name: string): Item {
        return this.chatItems.find((curTheme) => curTheme.name === name) as Item;
    }
}

export const DEFAULT_PREF: UserPreferences = {
    appThemeSelected: '',
    boardItems: [],
    boardItemSelected: new Item('Par défaut', 0, '', 'Composé de vert clair et de turquoise'),
    chatItems: [],
    chatItemSelected: new Item('Par défaut', 0, '', 'Composé de vert clair et de turquoise'),
    language: Language.French,
};
